File: My/1.UI/5.Intent/Intent.java
public class MainActivity
{
    // here are codes
    public void click(View v)
    {
        Intent intent = new Intent(this, second.class);
        EditText editText = (EditText)findViewById(R.id.edittext);
        String message = editText.getText().toString();
        intent.putExtra("Bayashat", message);
        startActivity(intent);

    }
}

public class second
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        // Here are codes
        Intent intent = getIntent();
        String message = intent.getStringExtra("Bayashat");
        TextView txt = findViewById(R.id.txt);
        txt.setText(message);

    }
}

File: My/1.UI/5.Intent/Intent.kt
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    // Here are codes
    val editText = findViewById<EditText>(R.id.edittext)
    val message = editText.text.toString()
    val intent = Intent(this, second::class.java).apply {
        putExtra("Bayashat", message)
    }
    startActivity(intent)

}

class Second : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    
    // Here are codes
    val message = intent.getStringExtra("Bayashat")
    val txt = findViewById<TextView>(R.id.txt)
    txt.setText(message)

}

